### ISSUE TITLE
Proper non-null handling

### ISSUE THREAD

**Issue by jexp** [neo4j-graphql/neo4j-graphql#80]
So here's the next thing. I'm now able to take IDL like in issue #77:

```
type Issue  {
        # issue number
        number: Float,
        repo: Repo! @relation(name: "BELONGS_TO"),
    }
type Repo  {
        owner: String,
        name: String,
}
```

and I'm able to run a query:

 ```
{ Issue { repo(name: "nonexistent") { name, owner } } }
```

this is progress. before, when repo was an optional field, I'd get all the issues back all the issues in the db, each with a null repo field.

Now, when the matching repo isn't found on the issue, the nullness rolls upward to the next optional thing, so I get an array of nulls, one for each issue in the database.

**Comment by jexp** [comment]
Regarding null-ness: https://medium.com/@calebmer/when-to-use-graphql-non-null-fields-4059337f6fc8

> This is a less obvious downside of non-null fields, but one you should consider nonetheless. Whenever an error happens in a non-null GraphQL field then that error is propagated up to the first nullable field. This means that small errors that may be constrained to a single field end up wiping out what may have been useful data.

**Comment by jessitron** [comment]
yes, I was trying to use it for exactly that functionality. I wanted the nullness to propagate upward, when a required field is selected into nullness. An issue without a repo doesn't make sense, so if I filter out its repo I don't want to see the issue at all.
